### DISCUSSION TITLE
Interpolated value should change the translation value, e.g. in French "pas de contacts" vs "pas d'élèves".

### DISCUSSION THREAD

**Discussion by david-bulte** [i18next/i18next#1840] [Q&A]
My question is twofold. First, imagine you have some "empty state" message, like "There are no contacts", "There are no patients" etc...

My first idea would be to use interpolation for this, e.g.

`empty-state = There are no $t({{item}}_plural)` (which you use like so: `i18next.t('empty_state', {item: 'contact'})`)

This however won't work in all languages, e.g. in French, this takes the following form:
_il n'y a pas de contacts_

Depending on the start of the term, the _de_ part can take another form:
_il n'y a pas d'élèves_ (there are no pupils),  _il n'y a pas d'hommes_ (there are no men), _il n'y a pas de héros_ (there are no heroes)

So the rule seems to be like so: _de_ is truncated to _d'_ in case it is followed by a vowel or a mute h (but _not_ when followed by a aspirated h)

So how would you handle this? My first thought was doing something with formatting but that seems like a dead end. Maybe using interpolation is not a good idea here, and I should bite the bullet and introduce a key for each individual case ('emtpy-state_contacts', 'empty-state_pupils' ... ?

And with that I have come to the second part of my question. At the moment of writing I have no idea into which languages the site is going to be translated (potentially a lot). How do you prepare for that? From my limited experience with this French translation, it seems that relying to heavily on interpolation could be dangerous?

**Comment by jamuhl** [comment]
The answer to this might be not what you like to hear: Don't concatenate string (use nesting $t only for terms you like to reuse - not for concatenating strings). It's one of the first rules in internationalization - do not concatenate strings as different languages handle this completely different like your samples show.

**Reply by david-bulte** [reply]
Thanks for your answer. The use of the nested $t might maybe have obscured the example. I think the problem comes as a result of interpolation. And now I'm wondering, taking that _do not concatenate strings_ rule into account, when would one use interpolation at all?

(Also, do you have some good pointers to documentation on the dos and donts of internationalization? Thanks!)

**Reply by jamuhl** [reply]
Interpolating for inserting things you do not enter yourself -> username, amounts, ... there are things that make sense 

your sample `empty-state = There are no $t({{item}}_plural)` is not really interpolating - it's trying to follow the developers dream of dry principle which does only work in programming language but not in real languages...

Your sample should be:

```js
contacts_zero = There are no contacts
contacts_one = There is one contact
contacts_other = There are {count} contacts
```

(using current version with updated plural forms - https://www.i18next.com/translation-function/plurals#singular-plural)
